commons/com.b2international.index/src/com/b2international/index/es/client/http/EsHttpClient.java

```diff
@@ -0,0 +1,179 @@
+					.setSocketTimeout(configuration.getSocketTimeout());
+			
+			final RestClientBuilder restClientBuilder = RestClient.builder(host)
+				.setRequestConfigCallback(requestConfigCallback)
+				.setMaxRetryTimeoutMillis(configuration.getSocketTimeout()); // retry timeout should match socket timeout
+			
+			if (configuration.isProtected()) {
+				
+				final HttpClientConfigCallback httpClientConfigCallback = httpClientConfigBuilder -> {
+					final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+					credentialsProvider.setCredentials(AuthScope.ANY, 
+							new UsernamePasswordCredentials(configuration.getUserName(), configuration.getPassword()));
+					return httpClientConfigBuilder.setDefaultCredentialsProvider(credentialsProvider);
+				};
+				
+				restClientBuilder.setHttpClientConfigCallback(httpClientConfigCallback);
+				
+			}
+			
+			final RestHighLevelClient client = new RestHighLevelClient(restClientBuilder);
+			
+			try {
+				checkState(client.ping(RequestOptions.DEFAULT), "The cluster at '%s' is not available.", host.toURI());
+			} catch (Exception e) {
+				if (e instanceof ElasticsearchStatusException && ((ElasticsearchStatusException) e).status() == RestStatus.UNAUTHORIZED) {
+					EsClient.LOG.error("Unable to authenticate with remote cluster '{}' using the given credentials", host.toURI());
+				}
+				close();
+				throw e;
+			}
+			
+			return client;
+		});
+		
+		this.clientExt = new RestHighLevelClientExt(client);
+		this.indicesClient = new IndicesHttpClient(client);
+		this.clusterClient = new ClusterHttpClient(client);
+	}
+
+	@Override
+	public final void close() throws IOException {
+		client.close();
+	}
+	
+	@Override
+	public Builder bulk(Listener listener) {
+		return BulkProcessor.builder((req, actionListener) -> clientExt.bulkAsync(req, RequestOptions.DEFAULT, actionListener), listener);
+	}
+
+
+	@Override
+	public final IndicesClient indices() {
+		return indicesClient;
+	}
+
+	@Override
+	public final ClusterClient cluster() {
+		return clusterClient;
+	}
+
+	@Override
+	public GetResponse get(GetRequest req) throws IOException {
+		return client.get(req, RequestOptions.DEFAULT);
+	}
+
+	@Override
+	public SearchResponse search(SearchRequest req) throws IOException {
+		return client.search(req, RequestOptions.DEFAULT);
+	}
+	
+	@Override
+	public SearchResponse scroll(SearchScrollRequest req) throws IOException {
+		return client.scroll(req, RequestOptions.DEFAULT);
+	}
+	
+	@Override
+	public final ClearScrollResponse clearScroll(ClearScrollRequest req) throws IOException {
+		return clientExt.clearScroll(req, RequestOptions.DEFAULT);
+	}
+	
+	@Override
+	public BulkByScrollResponse updateByQuery(String index, String type, int batchSize, Script script, int numberOfSlices, QueryBuilder query)
+			throws IOException {
+		UpdateByQueryRequest updateByQueryRequest = new UpdateByQueryRequest(new SearchRequest());
```

Comment:
The zero-arg constructor also starts out with an unpopulated `SearchRequest`: https://github.com/elastic/elasticsearch/blob/bc5b1afb55c3b9a75c8d3f69c7832765918b439c/server/src/main/java/org/elasticsearch/index/reindex/UpdateByQueryRequest.java#L47-L49

Reply:
Bumped to ES 6.5.0 in the meantime, which already got rid of the mentioned constructor.